src/riak_pipe.app.src

```diff
@@ -7,7 +7,10 @@
                   kernel,
                   stdlib,
                   sasl,
-                  riak_core
+                  riak_core,
+                  cluster_info,
+                  exometer_core,
+                  dialyzer
```

Comment:
not really running dialyzer in production??
Why is this in the list of applications, I wonder.

Reply:
Perhaps there's a better way of doing what its doing.

The `riak_pipe_v:type_of/1` function is used to report the type of what has been passed in when they type is not as expected.  For example, in an error like this:

```
validate_nval(NVal) ->
    {error, io_lib:format(
              "expected a positive integer,"
              " or a function or {Mod, Fun} of arity 1; not a ~p",
              [riak_pipe_v:type_of(NVal)])}.
```

To achieve this it uses functions exported from dialyzer - so now with OTP26 dialyzer needs to be in the list of applications or dialyzer will complain you're using a function that has not been included.

It would be possible to just stop reporting the type back in those error logs, but is that worse than including dialyzer in the applications?

Or perhaps I could just do my own function that uses the lis_list/1 is_binary/1 is_tuple/1 etc guards rather than relying on dialyzer?

Reply:
Changed the function, and remove dialyzer from the application list.